Sources/TuistSigning/SigningCipher.swift

```diff
@@ -38,47 +38,121 @@ enum SigningCipherError: FatalError, Equatable {
 /// SigningCiphering handles all encryption/decryption of files needed for signing (certificates, profiles, etc.)
 public protocol SigningCiphering {
     /// Encrypts all signing files at `Tuist/Signing`
-    func encryptSigning(at path: AbsolutePath) throws
-    /// Decrypts all signing files at `Tuist/Signing`
-    func decryptSigning(at path: AbsolutePath) throws
+    /// - Parameters:
+    ///     - keeepFiles: Keep unencrypted files
+    func encryptSigning(at path: AbsolutePath, keepFiles: Bool) throws
+    /// Decrypts all signing files at `Tuist/Signing
+    /// - Parameters:
+    ///     - keeepFiles: Keep encrypted files
+    func decryptSigning(at path: AbsolutePath, keepFiles: Bool) throws
+}
+
+public extension SigningCiphering {
+    func encryptSigning(at path: AbsolutePath) throws {
+        try encryptSigning(at: path, keepFiles: false)
+    }
+
+    func decryptSigning(at path: AbsolutePath) throws {
+        try decryptSigning(at: path, keepFiles: false)
+    }
 }
 
 public final class SigningCipher: SigningCiphering {
     private let rootDirectoryLocator: RootDirectoryLocating
+    private let signingFilesLocator: SigningFilesLocating
 
     /// Public initializer
-    public init(rootDirectoryLocator: RootDirectoryLocating = RootDirectoryLocator()) {
+    public convenience init() {
+        self.init(rootDirectoryLocator: RootDirectoryLocator(),
+                  signingFilesLocator: SigningFilesLocator())
+    }
+
+    init(rootDirectoryLocator: RootDirectoryLocating,
+         signingFilesLocator: SigningFilesLocating) {
         self.rootDirectoryLocator = rootDirectoryLocator
+        self.signingFilesLocator = signingFilesLocator
     }
 
-    public func encryptSigning(at path: AbsolutePath) throws {
-        let (signingKeyFiles, masterKey) = try signingData(at: path)
+    public func encryptSigning(at path: AbsolutePath, keepFiles: Bool) throws {
+        let masterKey = try self.masterKey(at: path)
+        let signingKeyFiles = try signingFilesLocator.locateUnencryptedSigningFiles(at: path)
+        guard !signingKeyFiles.isEmpty else { return }
         let cipheredKeys = try signingKeyFiles
             .map(FileHandler.shared.readFile)
             .map { try encryptData($0, masterKey: masterKey) }
 
-        try zip(cipheredKeys, signingKeyFiles).forEach {
-            logger.debug("Encrypting \($1.pathString)")
-            try $0.write(to: $1.url)
+        let correctlyEncryptedSigningFiles = try self.correctlyEncryptedSigningFiles(at: path, masterKey: masterKey)
+
+        try signingFilesLocator.locateEncryptedSigningFiles(at: path)
+            .filter { !correctlyEncryptedSigningFiles.contains($0) }
+            .forEach(FileHandler.shared.delete)
+
+        try zip(cipheredKeys, signingKeyFiles)
+            .filter { _, file in !correctlyEncryptedSigningFiles.contains(file) }
+            .forEach {
+                logger.debug("Encrypting \($1.pathString)")
```

Comment:
I think it would be worth creating named variables for these parameters - I typically only use the shorthand syntax if it will compact onto a single line and I don't need to keep any context of what's going on.

Reply:
Good catch, I should be more wary of this one 🙇